File: front-veterinaria/src/app/Services/rol.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { Rol, RolCustom, RolMapper } from '../Models/rol';
import { API_URL } from '../Environments/routes';

@Injectable({
  providedIn: 'root'
})
export class RolService {

  constructor(private http: HttpClient) { }

  findAll(): Observable<Rol[]> {
    return this.http.get<Rol[]>(API_URL.roles.find_all);
  }

  findAllCustom(): Observable<RolCustom[]> {
    return this.http.get<RolCustom[]>(API_URL.roles.find_all_custom);
  }

  findAllMapper(): Observable<RolMapper[]> {
    return this.http.get<RolMapper[]>(API_URL.roles.find_all_mapper);
  }

  findById(id: number): Observable<Rol> {
    return this.http.get<Rol>(API_URL.roles.find_by_id(id));
  }

  insert(rol: Rol): Observable<any> {
    return this.http.post<any>(API_URL.roles.insert, rol);
  }

  update(id: number, rol: Rol): Observable<any> {
    return this.http.put<any>(API_URL.roles.update(id), rol);
  }

  delete(id: number): Observable<any> {
    return this.http.delete<any>(API_URL.roles.delete(id));
  }
}

File: back-veterinaria/src/main/java/idat/proyecto/veterinaria/service/RolService.java
package idat.proyecto.veterinaria.service;

import org.springframework.http.ResponseEntity;

import idat.proyecto.veterinaria.entity.Rol;

public interface RolService {
	
	public abstract ResponseEntity<?> insert(Rol rol);
	public abstract ResponseEntity<?> update(Integer id, Rol rol);
	public abstract ResponseEntity<?> delete(Integer id);
	public abstract ResponseEntity<?> findById(Integer id);
	public abstract ResponseEntity<?> findAll();
	public abstract ResponseEntity<?> findAllCustom();
	public abstract ResponseEntity<?> findAllMapper();
}
